src/NewTools-Debugger/StDebuggerExtensionInspectorNodeBuilderVisitor.class.st

```diff
@@ -0,0 +1,41 @@
+"
+I am a debugger extension visitor that is used to display in the debugger inspector variables that were added via debugger extensions.
+"
+Class {
+	#name : #StDebuggerExtensionInspectorNodeBuilderVisitor,
+	#superclass : #StDebuggerExtensionVisitor,
+	#category : #'NewTools-Debugger-Model'
+}
+
+{ #category : #'instance creation' }
+StDebuggerExtensionInspectorNodeBuilderVisitor >> newCollection [
+
+	^ #(  ) asOrderedCollection
+]
+
+{ #category : #visiting }
+StDebuggerExtensionInspectorNodeBuilderVisitor >> visit: aStDebuggerExtension [
+
+	| extensionNodes |
+	extensionNodes := super visit: aStDebuggerExtension.
+	"If the visit hook that is used does nothing, then we return an empty collection"
+	^ extensionNodes == self ifTrue: [ self newCollection ] ifFalse: [ extensionNodes ]
```

Comment:
Prefer to store the results into an inst var rather than returning data for which we cannot predict the type.